File: dataset/human/c/file_1842.c
#include<stdio.h>

int main(void){
    
    int N,A[20],B[20],C[20],i,ans = 0;

    scanf("%d",&N);
    for(i = 0;i < N;i++){
        scanf("%d",&A[i]);
    }
    for(i = 0;i < N;i++){
        scanf("%d",&B[i]);
    }
    for(i = 0;i < N - 1;i++){
        scanf("%d",&C[i]);
    }

    for(i = 0;i < N;i++){
        ans += B[A[i]-1];
        //printf("%d 追加 ans:%d\n",B[A[i]-1],ans);
        if(A[i] -  1 == A[i - 1]){
            ans += C[A[i - 1]-1];
            //printf("Cの %d 追加 ans:%d\n",C[A[i-1]-1],ans);
        }
    }

    printf("%d\n",ans);
    return 0;
}

File: dataset/human/java/file_1842.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        // write your code here
        Scanner sc = new Scanner(System.in);

        int d = sc.nextInt();
        int g = sc.nextInt();
        int[][] scoreList = new int[d][3];

        for (int i = 0; i < d; i++) {
            scoreList[i][0] = sc.nextInt();
            scoreList[i][1] = sc.nextInt();
            scoreList[i][2] = scoreList[i][0] * (100 * (i + 1)) + scoreList[i][1];
        }

        int ans = Integer.MAX_VALUE;

        for (int i = 0; i < Math.pow(2, d); i++) {
            int nowBit = i;
            int index = Integer.highestOneBit(nowBit);
            int pointSum = 0;
            int count = 0;
            while (index > 0) {
                nowBit -= index;
                index = (int)(Math.log(index) / Math.log(2));
                pointSum += scoreList[index][2];
                count += scoreList[index][0];
                index = Integer.highestOneBit(nowBit);
            }
            if (pointSum < g) {
                int amari = g - pointSum;
                int maxIndex = (int)(Math.log(Integer.highestOneBit(~i & ((int) Math.pow(2, d) - 1))) / Math.log(2));
                int value =  (maxIndex + 1) * 100;
                if (amari <= (scoreList[maxIndex][0] - 1) * value) {
                    count += Math.ceil((double)amari / value);
                } else {
                    continue;
                }
            }
            ans = Math.min(ans, count);
        }

        System.out.println(ans);
    }
}
